docs/internal-documentation/maintenance/ReleaseProcess-deb-packages.md

```diff
@@ -0,0 +1,113 @@
+Release Process - Create .deb Packages
+=====================================
+
+
+## Overview
+
+The purpose of this process is to create and publish .deb packages for
+Zonemaster components, currently only Zonemaster-CLI, Zonemaster-Engine and
+Zonemaster-LDNS are covered.
+
+The process to build the packages is split in two parts. First "nightly
+packages" are updated, those packages are based on the develop branch
+of Zonemaster git repositories. Then they are promoted to "stable packages".
+The only difference between a nightly packages and its promoted version is the
+upstream tarball. While the nightly packages use the develop branch as upstream,
+the stable ones use a specific tag.
+
+## 1. Update the nightly packages
+
+The sources for nightly packages are kept in the branch `<distribution>-nightly`.
+
+For each package:
+
+1. Update the `debian/control` file to add and/or remove any dependencies that
+   have changed.
+
+2. If there are new files (executables as such), update `debian/*.manpages` and
+   `debian/*.install` accordingly. This step is not required for new Perl
+   modules, but will be for new scripts.
+
+## 2. Promote the nightly build to stable
+
+After the Github release and when the nightly package is working fine, promote
+it to stable:
+
+1. Locally merge `<distribution>-nightly` into `<distribution>`.
+
+2. Update the upstream ref in `pkg.sh` with the tag of the corresponding release
+   for each package.
+
+3. Update the `changelog` for each packages, by adding a new entry. The first line
+   should contain the package version in format
+   `<project version>-<package version>+deb<debian version>`.
+
+   For example:  `zonemaster-cli (3.1.0-3+deb11)` is the partial entry for
+   Zonemaster CLI version `3.1.0`, the package version is `3` and it is built
+   for Debian 11.
+
+4. Push the modifications to the packages sources repository. Then the packages
+   are automatically built and deployed to package.zonemaster.net.
+
+> Note: It is fairly important to perform all of those steps in "one push"
+> as the continuous deployment pipeline will start building the new packages
+> on the push event.
+
+
+## 3. Test the packages
+
+To test the newly created packages you can configure Zonemaster packages
+repository using:
+
+```sh
+curl -Ls https://package.zonemaster.net/setup.sh | sudo sh
```

Comment:
Have you considered also adding the `-S` and `-f` flags?

Reply:
Done